react-components/src/components/App/app.js

```diff
@@ -11,90 +11,24 @@
-
-const generateCacheData = introspectionQueryResultData => {
-    const fragmentMatcher = new IntrospectionFragmentMatcher({
-        introspectionQueryResultData
-    });
-    return new InMemoryCache({ fragmentMatcher });
-};
-
-const sessionStorage = window.sessionStorage;
-let graphQlFragmentTypes = sessionStorage.getItem('graphQlFragmentTypes');
+import { graphqlAuthLink } from '../../utils/authUtils';
 
 const App = props => {
     const { graphqlEndpoint, storeView = 'default' } = useConfigContext();
 
     const clientConfig = {
-        cache: graphQlFragmentTypes !== null ? generateCacheData(JSON.parse(graphQlFragmentTypes)) : undefined,
-        uri: graphqlEndpoint,
-        headers: { Store: storeView },
-        request: operation => {
-            let token = checkCookie('cif.userToken') ? cookieValue('cif.userToken') : '';
-            if (token.length > 0) {
-                operation.setContext({
-                    headers: {
-                        authorization: `Bearer ${token && token.length > 0 ? token : ''}`
-                    }
-                });
-            }
-        }
+        link: from([graphqlAuthLink, new HttpLink({ uri: graphqlEndpoint, headers: { Store: storeView } })]),
+        cache: new InMemoryCache()
     };
 
-    const [client, setClient] = useState(new ApolloClient(clientConfig));
-
-    useEffect(() => {
-        if (graphQlFragmentTypes === null) {
-            fetch(graphqlEndpoint, {
-                method: 'POST',
-                headers: { 'Content-Type': 'application/json' },
-                body: JSON.stringify({
-                    variables: {},
-                    query: `
-                        {
-                            __schema {
-                                types {
-                                kind
-                                name
-                                possibleTypes {
-                                    name
-                                }
-                                }
-                            }
-                        }
-                    `
-                })
-            })
-                .then(result => result.json())
-                .then(result => {
-                    // here we're filtering out any type information unrelated to unions or interfaces
-                    const filteredData = result.data.__schema.types.filter(type => type.possibleTypes !== null);
-                    result.data.__schema.types = filteredData;
-
-                    graphQlFragmentTypes = JSON.stringify(result.data);
-                    sessionStorage.setItem('graphQlFragmentTypes', graphQlFragmentTypes);
-
-                    setClient(
-                        new ApolloClient({
-                            ...clientConfig,
-                            cache: generateCacheData(result.data)
-                        })
-                    );
-                });
-        }
-    });
+    const client = new ApolloClient(clientConfig);
```

Comment:
Isn't this `client` supposed to come from the custom application?

Reply:
I assume, if a customer uses their own `ApolloClient`, they wouldn't use `CommerceApp` anymore, but add their own `ApolloProvider` together with a mix of their own and our contexts.

Reply:
But this defies the purpose of the `CommerceApp`. As soon as they want to have a custom component that wants to fetch data they have to refactor all their code to drop `CommerceApp` and re-create our provider stack.

Reply:
I can modify `CommerceApp` to receive the client as a `prop`. if that prop is not provided, then a default client will be created.